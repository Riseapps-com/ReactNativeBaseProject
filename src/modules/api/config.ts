const AXIOS_REQUEST_TIMEOUT_MSEC = 16000;

export default AXIOS_REQUEST_TIMEOUT_MSEC;
