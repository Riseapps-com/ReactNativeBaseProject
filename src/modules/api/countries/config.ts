export const ALL = 'all';
export const BY_REGION = 'region';
export const BY_CODE = 'alpha';
