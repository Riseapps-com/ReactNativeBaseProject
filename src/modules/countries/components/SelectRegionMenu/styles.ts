import { sizes, styleSheetFactory } from '~theme';

export default styleSheetFactory(() => ({
  container: {
    flex: 1,

    marginVertical: sizes.SPACING_S,
  },
}));
