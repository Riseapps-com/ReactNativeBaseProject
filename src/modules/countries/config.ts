import { LocalRegion } from './types';

export const STORYBOOK_SCREEN_NAME = 'STORYBOOK_SCREEN';
export const MENU_SCREEN_NAME = 'MENU_SCREEN';
export const COUNTRIES_SCREEN_NAME = 'COUNTRIES_SCREEN';
export const SELECT_REGION_SCREEN_NAME = 'SELECT_REGION_SCREEN';
export const COUNTRY_DETAILS_SCREEN_NAME = 'COUNTRY_DETAILS_SCREEN';

export const regions: LocalRegion[] = ['africa', 'americas', 'asia', 'europe', 'oceania'];
