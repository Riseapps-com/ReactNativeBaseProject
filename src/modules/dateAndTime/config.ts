export const LOG_DATE_TIME_FORMAT = 'YYYY-MM-DD HH:mm:ss.SSSZ';
