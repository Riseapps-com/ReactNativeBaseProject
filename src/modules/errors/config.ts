import type { ErrorCode } from './types';

export const RETRYABLE_ERROR_CODES: ErrorCode[] = ['UnknownException', 'ServerErrorException'];
