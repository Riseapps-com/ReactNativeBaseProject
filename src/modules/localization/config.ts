export const LOCALIZATION_FALLBACK_LANGUAGE = 'en';
