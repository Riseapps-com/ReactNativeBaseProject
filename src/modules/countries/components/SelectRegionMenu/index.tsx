import { useNavigation } from '@react-navigation/native';
import React, { useCallback } from 'react';
import { useTranslation } from 'react-i18next';
import { View } from 'react-native';

import { useImages } from '~assets';
import { testIDs } from '~config';
import { useTheme } from '~theme';

import { COUNTRIES_SCREEN_NAME, regions } from '../../config';
import { SelectRegionNavigation } from '../../types';
import MenuItem from '../MenuItem';
import themedStyles from './styles';

const SelectRegionMenu: React.FC = () => {
  const { t } = useTranslation();
  const navigation = useNavigation<SelectRegionNavigation>();
  const [styles] = useTheme(themedStyles);
  const images = useImages();

  const handleItemPress = useCallback(
    async (index: number) => {
      navigation.navigate(COUNTRIES_SCREEN_NAME, {
        region: regions[index],
        title: t(regions[index]),
      });
    },
    [navigation, t]
  );

  return (
    <View style={styles.container}>
      <MenuItem
        title={t('africa')}
        image={images.africa}
        index={0}
        onItemPress={handleItemPress}
        testID={testIDs.selectRegion.africa}
      />

      <MenuItem
        title={t('americas')}
        image={images.americas}
        index={1}
        onItemPress={handleItemPress}
        testID={testIDs.selectRegion.americas}
      />

      <MenuItem
        title={t('asia')}
        image={images.asia}
        index={2}
        onItemPress={handleItemPress}
        testID={testIDs.selectRegion.asia}
      />

      <MenuItem
        title={t('europe')}
        image={images.europe}
        index={3}
        onItemPress={handleItemPress}
        testID={testIDs.selectRegion.europe}
      />

      <MenuItem
        title={t('oceania')}
        image={images.oceania}
        index={4}
        onItemPress={handleItemPress}
        testID={testIDs.selectRegion.oceania}
      />
    </View>
  );
};

export default SelectRegionMenu;
