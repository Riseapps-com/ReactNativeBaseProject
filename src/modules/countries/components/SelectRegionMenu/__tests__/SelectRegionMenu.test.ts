import { fireEvent } from '@testing-library/react-native';

import { testIDs } from '~config';
import { I18n } from '~modules/localization';
import { mockedUseNavigation, renderComponent } from '~modules/tests';

import { COUNTRIES_SCREEN_NAME } from '../../../config';
import SelectRegionMenu from '../index';

const renderSelectRegionMenu = () => renderComponent(SelectRegionMenu);

describe('countries', () => {
  describe('<SelectRegionMenu />', () => {
    it('render <SelectRegionMenu />', () => {
      const selectRegionMenu = renderSelectRegionMenu();

      expect(selectRegionMenu.getByTestId(testIDs.selectRegion.africa)).toBeTruthy();
      expect(selectRegionMenu.getByTestId(testIDs.selectRegion.americas)).toBeTruthy();
      expect(selectRegionMenu.getByTestId(testIDs.selectRegion.asia)).toBeTruthy();
      expect(selectRegionMenu.getByTestId(testIDs.selectRegion.europe)).toBeTruthy();
      expect(selectRegionMenu.getByTestId(testIDs.selectRegion.oceania)).toBeTruthy();
      expect(selectRegionMenu).toMatchSnapshot();
    });

    it('navigates to countries', async () => {
      const selectRegionMenu = renderSelectRegionMenu();

      fireEvent.press(selectRegionMenu.getByTestId(testIDs.selectRegion.africa));
      jest.runAllTimers();

      expect(mockedUseNavigation().navigate).toBeCalledTimes(1);
      expect(mockedUseNavigation().navigate).toBeCalledWith(COUNTRIES_SCREEN_NAME, {
        region: 'africa',
        title: I18n.t('africa'),
      });
    });
  });
});
