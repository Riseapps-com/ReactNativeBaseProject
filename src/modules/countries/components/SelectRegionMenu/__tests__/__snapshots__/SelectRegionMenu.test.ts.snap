// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`countries <SelectRegionMenu /> render <SelectRegionMenu /> 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "marginVertical": 8,
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flex": 1,
        "marginHorizontal": 8,
        "marginVertical": 8,
        "opacity": 1,
        "padding": 8,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.25,
        "shadowRadius": 3.84,
      }
    }
    testID="selectRegion_africa"
  >
    <View
      style={
        Object {
          "backgroundColor": "#0036A7",
          "borderRadius": 8,
          "flex": 1,
          "flexDirection": "row",
          "overflow": "hidden",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "paddingHorizontal": 8,
            "width": "30%",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "height": 48,
                "width": 48,
              },
            ]
          }
        >
          <FastImageView
            resizeMode="cover"
            source={
              Object {
                "testUri": "../../../assets/images/africa.png",
              }
            }
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          />
        </View>
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "center",
            "paddingEnd": 16,
            "paddingStart": 8,
          }
        }
      >
        <Text
          numberOfLines={2}
          style={
            Array [
              Object {
                "color": "#4A4E4D",
              },
              Object {
                "fontFamily": "Quicksand-Bold",
              },
              Object {
                "fontSize": 16,
              },
              Object {
                "color": "#FFFFFF",
              },
            ]
          }
        >
          africa
        </Text>
      </View>
    </View>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flex": 1,
        "marginHorizontal": 8,
        "marginVertical": 8,
        "opacity": 1,
        "padding": 8,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.25,
        "shadowRadius": 3.84,
      }
    }
    testID="selectRegion_americas"
  >
    <View
      style={
        Object {
          "backgroundColor": "#0036A7",
          "borderRadius": 8,
          "flex": 1,
          "flexDirection": "row",
          "overflow": "hidden",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "paddingHorizontal": 8,
            "width": "30%",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "height": 48,
                "width": 48,
              },
            ]
          }
        >
          <FastImageView
            resizeMode="cover"
            source={
              Object {
                "testUri": "../../../assets/images/america.png",
              }
            }
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          />
        </View>
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "center",
            "paddingEnd": 16,
            "paddingStart": 8,
          }
        }
      >
        <Text
          numberOfLines={2}
          style={
            Array [
              Object {
                "color": "#4A4E4D",
              },
              Object {
                "fontFamily": "Quicksand-Bold",
              },
              Object {
                "fontSize": 16,
              },
              Object {
                "color": "#FFFFFF",
              },
            ]
          }
        >
          americas
        </Text>
      </View>
    </View>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flex": 1,
        "marginHorizontal": 8,
        "marginVertical": 8,
        "opacity": 1,
        "padding": 8,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.25,
        "shadowRadius": 3.84,
      }
    }
    testID="selectRegion_asia"
  >
    <View
      style={
        Object {
          "backgroundColor": "#0036A7",
          "borderRadius": 8,
          "flex": 1,
          "flexDirection": "row",
          "overflow": "hidden",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "paddingHorizontal": 8,
            "width": "30%",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "height": 48,
                "width": 48,
              },
            ]
          }
        >
          <FastImageView
            resizeMode="cover"
            source={
              Object {
                "testUri": "../../../assets/images/asia.png",
              }
            }
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          />
        </View>
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "center",
            "paddingEnd": 16,
            "paddingStart": 8,
          }
        }
      >
        <Text
          numberOfLines={2}
          style={
            Array [
              Object {
                "color": "#4A4E4D",
              },
              Object {
                "fontFamily": "Quicksand-Bold",
              },
              Object {
                "fontSize": 16,
              },
              Object {
                "color": "#FFFFFF",
              },
            ]
          }
        >
          asia
        </Text>
      </View>
    </View>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flex": 1,
        "marginHorizontal": 8,
        "marginVertical": 8,
        "opacity": 1,
        "padding": 8,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.25,
        "shadowRadius": 3.84,
      }
    }
    testID="selectRegion_europe"
  >
    <View
      style={
        Object {
          "backgroundColor": "#0036A7",
          "borderRadius": 8,
          "flex": 1,
          "flexDirection": "row",
          "overflow": "hidden",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "paddingHorizontal": 8,
            "width": "30%",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "height": 48,
                "width": 48,
              },
            ]
          }
        >
          <FastImageView
            resizeMode="cover"
            source={
              Object {
                "testUri": "../../../assets/images/europe.png",
              }
            }
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          />
        </View>
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "center",
            "paddingEnd": 16,
            "paddingStart": 8,
          }
        }
      >
        <Text
          numberOfLines={2}
          style={
            Array [
              Object {
                "color": "#4A4E4D",
              },
              Object {
                "fontFamily": "Quicksand-Bold",
              },
              Object {
                "fontSize": 16,
              },
              Object {
                "color": "#FFFFFF",
              },
            ]
          }
        >
          europe
        </Text>
      </View>
    </View>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flex": 1,
        "marginHorizontal": 8,
        "marginVertical": 8,
        "opacity": 1,
        "padding": 8,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.25,
        "shadowRadius": 3.84,
      }
    }
    testID="selectRegion_oceania"
  >
    <View
      style={
        Object {
          "backgroundColor": "#0036A7",
          "borderRadius": 8,
          "flex": 1,
          "flexDirection": "row",
          "overflow": "hidden",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "paddingHorizontal": 8,
            "width": "30%",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "height": 48,
                "width": 48,
              },
            ]
          }
        >
          <FastImageView
            resizeMode="cover"
            source={
              Object {
                "testUri": "../../../assets/images/oceania.png",
              }
            }
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          />
        </View>
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "center",
            "paddingEnd": 16,
            "paddingStart": 8,
          }
        }
      >
        <Text
          numberOfLines={2}
          style={
            Array [
              Object {
                "color": "#4A4E4D",
              },
              Object {
                "fontFamily": "Quicksand-Bold",
              },
              Object {
                "fontSize": 16,
              },
              Object {
                "color": "#FFFFFF",
              },
            ]
          }
        >
          oceania
        </Text>
      </View>
    </View>
  </View>
</View>
`;
